common/src/main/scala/services/teamwork/SquadService.scala

```diff
@@ -0,0 +1,3017 @@
+// Copyright (c) 2019 PSForever
+package services.teamwork
+
+import akka.actor.{Actor, ActorRef, Terminated}
+import net.psforever.objects.{Avatar, LivePlayerList, Player}
+import net.psforever.objects.definition.converter.StatConverter
+import net.psforever.objects.loadouts.SquadLoadout
+import net.psforever.objects.teamwork.{Member, Squad, SquadFeatures}
+import net.psforever.objects.zones.Zone
+import net.psforever.packet.game._
+import net.psforever.types._
+import services.{GenericEventBus, Service}
+
+import scala.collection.concurrent.TrieMap
+import scala.collection.mutable
+import scala.collection.mutable.ListBuffer
+
+class SquadService extends Actor {
+  import SquadService._
+
+  /**
+    * The current unique squad identifier, to be wrapped in a `PlanetSideGUID` object later.
+    * The count always starts at 1, even when reset.
+    * A squad of `PlanetSideGUID(0)` indicates both a nonexistent squad and the default no-squad for clients.
+    */
+  private var sid : Int = 1
```

Comment:
Do we have a strategy for when the squad service inevitably crashes and these SIDs reset to zero and new squads start to clobber old ones on the clients?

Reply:
Fix the `SquadService` so that it doesn't crash.
When this actor does stop, though, it does try to close all squads that were active at the time (see `postStop`).

Reply:
Gotcha. Akka offers some nice redundancy / error recovery mechanisms. We should look into those cases later on